### ISSUE TITLE
[feature] Add support for colored lights 

### ISSUE THREAD

**Issue by nnpanda** [andrei-tatar/node-red-contrib-nora#12]
Hello,
what about color support for light?

**Comment by andrei-tatar** [comment]
It's already present on the service, the node-red node needs to be updated to support it. PRs are welcomed! :)

**Comment by MrHarcombe** [comment]
I'm also looking at trying to get the colours working on neopixels connected to a Pi through this excellent service. What needs changing in node-red?

**Comment by Austrian4lex** [comment]
Hi, thank you for this amazing service! :)
I am actually also looking for the color support. (yes I do have the latest version installed)
What needs to change to make it work?
Thanks again

**Comment by andrei-tatar** [comment]
The light node needs a new checkbox and the rules (when it's visible or not based on the other checkboxes): https://github.com/andrei-tatar/node-red-contrib-nora/blob/master/src/nodes/nora-light.html
I already started working on this at some point (see the `light-color` id div). I might get some time to look over this in the following days.

**Comment by mariocarbone** [comment]
I'm waiting too for this function! :) i mounted the rgb led but i can't control the color with google

**Comment by Bhelma** [comment]
Hello,
I am using this node and I would like to be able to use the RGB of the lights.
I have updated Node network and nora to its latest versions.

How is this topic going?

**Comment by andrei-tatar** [comment]
Not too great. I usually implement only the things I also need because there's a lot of time needed to keep up (I would also like to add support for local execution).
Contributions are greatly appreciated! :)

**Comment by FernandoWahl** [comment]
Hello,


I'm trying to implement colors for NORA on node-red, but I realized that on the service you only implemented HSV (https://developers.google.com/assistant/smarthome/traits/colorsetting#rgb,-hsv,-and-color -temperature) Any reason not to enable RGB and temperature?

**Comment by andrei-tatar** [comment]
HSV can be converted to RGB in the plugin (or by some other node-red plugin). Just a bit of maths.
Temperature... not sure who of the DIY-ers need this... Probably most of us will play will colored RGBs. 
But feel free to add support to backend as well for RGB/temperature if you find it useful. I will have a look at the PR.

**Comment by FernandoWahl** [comment]
So I found another method to perform the core conversion, works very well. Actually, I'm using https://www.npmjs.com/package/color-convert with https://www.npmjs.com/package/node-red-contrib-function-npm. Working very well.

About HSV I didn't know this pattern before. Using here I found out more about it and realized that I can make some improvements later to optimize its use.

About PR, check the TODO that I put.

And I'm sorry for my google english translator :D

**Comment by andrei-tatar** [comment]
I will fix the service HSV vs Hsv. Didn't notice they used different caps. Kinda weird.
Thanks for noticing that! :)

**Comment by andrei-tatar** [comment]
PR merged

**Comment by Chrischi-** [comment]
next, we should integrate the color temperature.

**Comment by GravityRZ** [comment]
noticed 0.0.30 with color support and tried to get it to work(before i noticed things are not finished)

Hue works but S and V need to be selctable between 0-100%

also on the reame you state this

`Payload must be an object like { [on]: true/false, [brightness]: 0-100, [color]: { [spectrumHsv] : { [hue]: 0-360, [saturation]:0-1, [value]:0-1 } } }')`

i think those []  need to be replaced by ""

so correct syntax is

`Payload must be an object like { "on": true/false, "brightness": 0-100, "color": { "spectrumHsv" : { "hue": 0-360, "saturation":0-1, "value":0-1 } } }')`

took me a while before i noticed what was wrong

i get my Hue readings in Node-Red through MQTT so changing the color from b,cw,g,m,r,t,ww to HSL is a lot of work.
currently i only use the r,g,b, to convert to hsl and use the seperate brightness level but since the S,L part is not working yet i do not know if this is the way to go

**Comment by andrei-tatar** [comment]
Square brackets usually mean a field is optional. Of course JSON needs double quotes. However, javascript doesn't (if you write in the function node). 


```
return {
  payload: {
    on: false,
    brightness: 32,
    spectrumHsv: {
      hue: 31,
    },
  },
};
```

**Comment by GravityRZ** [comment]
yes i write in the function node but send it out as json (just like it is received as json. 

regarding the Sat and val. are you telling me those are not needed or not working yet?

this is what i send to the light(and what is send back)
without the saturation and value levels the color which goes in is not the same as the one going out(because sat and val get changed to 0 or 1 instead of a value between 0-100)

i noticed that another user has 31% on his sat level, how can this be?

`msg.payload = {
    "on": status,
    "brightness": brightness,
    "color": { "spectrumHsv" :
    { "hue": hue,
    "saturation": saturation,
    "value": value } } };
 return msg`

**Comment by GravityRZ** [comment]
@Sw0rdf1sh1 

should this not be closed.
